solutions/banking/watson-scripts/assistant-create.sh

```diff
@@ -2,16 +2,8 @@
 
 set -e
 
-WATSON_ASSISTANT_URL=$1
 project_name="gen-ai-rag-sample-app-assistant"
-
-# Expects the environment variable $IBMCLOUD_API_KEY to be set
-if [[ -z "${IAM_TOKEN}" ]]; then
-    echo "API key must be set with IAM_TOKEN environment variable" >&2
-    exit 1
-fi
-
-token="$(echo "$IAM_TOKEN" | awk '{print $2}')"
+token=$(curl -fLsS -X POST 'https://iam.cloud.ibm.com/identity/token' -H 'Content-Type: application/x-www-form-urlencoded' -d "grant_type=urn:ibm:params:oauth:grant-type:apikey&apikey=$API_KEY" | jq -r '.access_token')
 
 ASSISTANT_ID=$(curl --retry 3 -fLsS -X GET --location "$WATSON_ASSISTANT_URL/v2/assistants?version=2023-06-15" \
```

Comment:
The create script should do `exit 1` to indicate an error - otherwise it may try to complain about empty json (it ignores non-json lines at the start of the output)

Also you may want to add other attributes like integration ID retrieved after creation and merged with the data returned by create (which you might want to filter for some attributes that might change, like last update time)